test/double.js

```diff
@@ -68,3 +68,52 @@ describe("WebIDL double type", () => {
         assert.throws(() => sut(" 123,123 "), TypeError);
     });
 });
+
+describe("WebIDL unrestricted double type", () => {
+    var sut = conversions["double"];
+
+    it("should return `0` for `0`", () => {
+        assert.strictEqual(sut(0), 0);
+    });
+
+    it("should return `42` for `42`", () => {
+        assert.strictEqual(sut(42), 42);
+    });
+
+    it("should return `0` for `null`", () => {
+        assert.strictEqual(sut(null), 0);
+    });
+
+    it("should return `0` for `\"\"`", () => {
+        assert.strictEqual(sut(""), 0);
+    });
+
+    it("should return `0` for `false`", () => {
+        assert.strictEqual(sut(0), 0);
+    });
+
+    it("should return `1` for `true`", () => {
+        assert.strictEqual(sut(null), 0);
+    });
+
+    it("should return `0` for random whitespace", () => {
+        assert.strictEqual(sut(" \t\n\t "), 0);
+    });
+
+    it("should return `123` for `\" 123 \"`", () => {
+        assert.strictEqual(sut(" 123 "), 123);
+    });
+
+    it("should return `-123.123` for `\" -123.123 \"`", () => {
+        assert.strictEqual(sut(" -123.123 "), -123.123);
+    });
+
+    it("should not throw for any input", () => {
```

Comment:
Would you mind instead testing that these return the correct values? 0, 0, NaN, +Infinity, -Infinity, NaN in particular